Source/CarthageKit/Git.swift

```diff
@@ -217,14 +227,24 @@ public func cloneSubmoduleInWorkingDirectory(_ submodule: Submodule, _ workingDi
 				}
 		}
 
-	return SignalProducer<(), CarthageError> { () -> Result<(), CarthageError> in
-		repositoryCheck("remove submodule checkout") {
-			try FileManager.default.removeItem(at: submoduleDirectoryURL)
+	let dependencyForSubmodule = Dependency.git(submodule.url)
+
+	return cloneOrFetchOnComplete(dependency: dependencyForSubmodule, preferHTTPS: preferHTTPS, commitish: submodule.sha)
+		.flatMap(.merge) { repositoryURL -> SignalProducer<String, CarthageError> in
+			return SignalProducer<(), CarthageError> { () -> Result<(), CarthageError> in
+				repositoryCheck("remove submodule checkout") {
+					try FileManager.default.removeItem(at: submoduleDirectoryURL)
+				}
+			}
+			.then(cloneRepository(GitURL(repositoryURL.path), submoduleDirectoryURL, isBare: false))
 		}
-	}
-	.then(cloneRepository(submodule.url, workingDirectoryURL.appendingPathComponent(submodule.path), isBare: false))
-	.then(checkoutSubmodule(submodule, submoduleDirectoryURL))
-	.then(purgeGitDirectories)
+		.then(checkoutRepositoryToDirectory(submoduleDirectoryURL, submoduleDirectoryURL, force: false, revision: submodule.sha))
```

Comment:
Why is it important to have `force: false` here?

Reply:
I'm just trying to keep the original logic. I don't know why sometimes the checkout is forced sometimes is not.